pkgs/by-name/im/immich/machine-learning.nix

```diff
@@ -0,0 +1,85 @@
+{
+  lib,
+  src,
+  immich,
+  python3,
+}:
+python3.pkgs.buildPythonApplication {
```

Comment:
Casual reminder that the splicing-compatible way currently is `python3Packages`

Reply:
What is "splicing"?

Reply:
> what is "splicing"

The [cross-compilation machinery](https://nixos.org/manual/nixpkgs/unstable/#ssec-cross-dependency-implementation) that makes `mkDerivation` take `buildInputs` and `nativeBuildInputs` from different platforms.

> why do we want it here

It's part of nixpkgs functionality, best not to have it broken without reason

Reply:
```suggestion
python3Packages.buildPythonApplication {
```